hashintel/hash
# Allow setting property types before creating first entity type version

## 🌟 What is the purpose of this PR?




When you create an entity type, we now save this in a query param instead of immediately publishing, so that you can set properties on the first version

The caching changes have been agreed with @Alfred-Mountfield  and @benwerner01.  

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203007126736607/1203172443927344/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?




- Type system status is now shared between pages via a context, removing white flash when creating entity type
- Fixed a bug in the edit bar where it would change contents while animating away when version changed from 0 to 1
- Implemented discarding first version of an entity type
- Always display edit bar on version 0
- Don't hide entity type editor before property types have loaded, removing white flash when creating entity type
- `useEntityType` now has a function to let you publish the initial entity type version, and it won't clear the entity type when you do this, removing white flash when publishing first version of entity type
- Display data in entity type editor from draft query param when present instead of fetching, and mark this as version 0
- **getEntityType** block protocol method is now no longer cached, due to an issue with missing entity types returning an empty subgraph. This is so we can use getEntityType to check if a specified entity type title is unique, and to display an error if not. 

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



-

## ❓ How to test this?



1. Create entity type 
2. See data present in query param
3. Add some properties
4. Try publishing
5. Refresh, see those properties still present

-- 

1. Repeat steps 1-3
2. Try discarding
3. See that you're redirected to the create form

## 📹 Demo
